libs/ui/design-system/src/lib/input.tsx

```diff
@@ -1,28 +1,66 @@
-import type * as React from 'react'
-
-import { Input as InputPrimitive } from '@nmit-coursition/ui/primitives/input'
+import * as InputPrimitive from '@nmit-coursition/ui/primitives/input'
 import { Label } from '@nmit-coursition/ui/primitives/label'
 import { cn } from '@nmit-coursition/ui/utils'
+import { type VariantProps, cva } from 'class-variance-authority'
+
+type OverrideProps<T, V> = V & Omit<T, keyof V>
 
-type InputType = React.ComponentProps<'input'>['type']
+const inputVariants = cva('', {
+  variants: {
+    variant: {
+      primary: '',
+      verified: '', // green borders
+      unVerified: '', // red borders
+      disabled: '',
+    },
+  },
+})
 
-interface InputProps extends React.ComponentProps<'input'> {
+type Orientation = 'horizontal' | 'vertical'
```

Comment:
**suggestion:** Orientation **is** variant, it shouldn't be passed as different prop. That's what cva is for. I bet this is where we need `mixinProps`.

Reply:
So, we need to merge that proposal to the main branch to see what effect it has :)

Reply:
Will check it out and merge. :)

Reply:
Also I am not sure... do we have horizontal / vertical in the design system? Now I don't see them, but in the past they were there, so I asked the designer.

Reply:
I was referring to the old primitive page, haha